src/SonarScanner.MSBuild.Tasks/WriteProjectInfoFile.cs

```diff
@@ -340,7 +340,10 @@ private bool TryGetSettingValue(ITaskItem taskItem, out string metadataValue)
                     .FirstOrDefault();
             }
 
-            return null;
+            Log.LogMessage(Resources.WPIF_GeneratingRandomGuid, FullProjectPath);
+
+            //Generating a new guid for projects without one
+            return Guid.NewGuid().ToString();
```

Comment:
What impact is this going to have for users who are still on older versions of SQ i.e. pre-7.6? I can see this generating potentially time consuming support/community issues.

Would a safer option be to preserve the old behaviour for versions of SQ prior to 7.6?

Reply:
Well, as far as i remember, we're going to only keep compatibility with the last LTS, not below, on this version, so i'm not sure we need to do that. WDYT ?

Reply:
I don't know what the compatibility strategy is for the scanners, I'm afraid. I don't know if it is the same as support for SQ.

How is the user going to know that their version of SQ is unsupported?
I can't see anything in our documentation that explains what the relationship between the scanners and versions of SQ is e.g. a compatibility matrix. Even if it is documented, I doubt many customers will read it until after they have a problem and they're trying to work out what went wrong.

I think the current code could give rise to community/support issues that we could avoid, either by preserving the old behaviour for older SQ versions, or having the scanner explicitly fail in the `begin` step if the version of SQ is too old.

 The scanner already contains some conditional logic based on the SQ version (see e.g. `GetAnalyzerSettings`), so implementing either check should be straightforward.